lib/schedule.js

```diff
@@ -14,6 +16,12 @@ var defaultClock = {
 
 var clock = defaultClock;
 
+var modes = {
```

Comment:
Essentially the modes are:
1. ack - we always want to call this task `ASAP`,
2. checkReclaim - this we should `RESCHEDULE` until the host is under load, and
3. tryReclaim - this should be `ABANDON` ed if the host is under load

Reply:
Also processHead needs ASAP